android-components/components/browser/engine-gecko/src/main/java/mozilla/components/experiment/NimbusExperimentDelegate.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.experiment
+
+import mozilla.components.browser.engine.gecko.GeckoNimbus
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
+import org.mozilla.geckoview.ExperimentDelegate
+import org.mozilla.geckoview.GeckoResult
+
+/**
+ * Default Nimbus [ExperimentDelegate] implementation to communicate with mobile Gecko and GeckoView.
+ */
+class NimbusExperimentDelegate : ExperimentDelegate {
+
+    private val logger = Logger(NimbusExperimentDelegate::javaClass.name)
+
+    /**
+     * Retrieves experiment information on the feature for use in GeckoView.
+     *
+     * @param feature Nimbus feature to retrieve information about
+     * @return a JSON object containing experiment information
+     */
+    override fun onGetExperimentFeature(feature: String): GeckoResult<JSONObject>? {
+        val result = GeckoResult<JSONObject>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
+        if (nimbusFeature != null) {
+            result.complete(nimbusFeature.toJSONObject())
+        } else {
+            logger.warn("Could not find Nimbus feature '$feature' to retrieve experiment information.")
```

Comment:
I wonder if this should be more an error than a warning— if you have feature correct, then you should get something back. Otherwise, the caller has got the feature name incorrect, and you should likely fail fast as it'd be a developer error. I don't know the error philosophy in GeckoView.

Reply:
Good idea, thanks, adjusted to use an `ExperimentException` for when things go wrong!

Is it possible for me to check for valid`slugs` as well?

Reply:
For slugs, no there isn't. The slug is an opaque string that you get from the feature configuration ([for example](https://experimenter.info/messaging/mobile-messaging#experimenting-with-messages)). If there is no experiment, then feature authors won't be calling the method.